transformer_engine/pytorch/tensor/float8_tensor.py

```diff
@@ -780,23 +778,21 @@ def transpose_2d(
                 fill_cache = False
 
         # Need to compute transpose if cache is invalid
-        need_compute = force_compute
-        if self._transpose is None:
-            need_compute = True
-        elif self._transpose_invalid:
-            need_compute = True
-
-        # Need to apply transpose kernel if noop flag is applied
-        if noop_flag is not None:
-            need_compute = True
+        need_compute = (
+            force_compute
+            or (self._transpose is None)
+            or self._transpose_invalid
+            or (noop_flag is not None)
+        )
 
         # Return cached transpose if possible
         if not need_compute:
+            assert self._transpose is not None
             return self._transpose
 
         # Allocate output if needed
         data = self._data.contiguous().reshape(-1, self.size(-1))
-        out = self._transpose
+        out: Optional[torch.Tensor] = self._transpose
```

Comment:
Would pyright complain about the following?
```suggestion
        out: torch.Tensor = self._transpose
```
This matches the intent more closely, although it might temporarily be incorrect.

Reply:
It would since the very next line is `if out is None`, which would be useless if out had to be a Tensor :-).